src/projects/projects.service.ts

```diff
@@ -19,14 +19,25 @@ export class ProjectsService {
     private readonly projectRepository: Repository<Project>,
   ) {}
 
-  async findOrFail(id: string, clientId: string): Promise<Project> {
+  async findOrFail(
+    id: string,
+    clientOrWorkspaceId: string,
+    type: 'client' | 'workspace' = 'client',
+  ): Promise<Project> {
     if (!isUUID(id)) {
       throw new BadRequestException('Invalid projectId format')
     }
 
+    let whereClause: Record<string, any>
+    if (type === 'client') {
```

Comment:
in case you are working with such fixed text that have a particular meaning and have to meet a certain condition. I suggest using enums